lib/metrics/normalizer.js

```diff
@@ -112,45 +110,53 @@ MetricNormalizer.prototype.load = function load(json) {
   }
 }
 
+/**
+ *  Helper for loadFromConfig
+ *
+ * @param rule {object} from this.config.rules
+ * @param ctx {object} context from the calling function
+ */
+const processNameRule = function processName(rule, ctx) {
+  if (!rule.pattern) {
+    return logger.error({ rule: rule }, 'Simple naming rules require a pattern.')
+  }
+  if (!rule.name) {
+    return logger.error({ rule: rule }, 'Simple naming rules require a replacement name.')
+  }
+
+  const precedence = rule.precedence
+  const terminal = rule.terminate_chain
+  const json = {
+    match_expression: rule.pattern,
+    eval_order: typeof precedence === 'number' ? precedence : 500,
+    terminate_chain: typeof terminal === 'boolean' ? terminal : true,
+    replace_all: rule.replace_all,
+    replacement: rule.name,
+    ignore: false
+  }
+
+  // Find where the rule should be inserted and do so.
+  const reverse = ctx.config.feature_flag.reverse_naming_rules
+  const insert = ctx.rules.findIndex(function findRule(r) {
+    return reverse ? r.precedence >= json.eval_order : r.precedence > json.eval_order
+  })
+  if (insert === -1) {
+    ctx.rules.push(new NormalizerRule(json))
+  } else {
+    ctx.rules.splice(insert, 0, new NormalizerRule(json))
+  }
+}
 /**
  * Load any rules found in the configuration into a metric normalizer.
  *
  * Operates via side effects.
  */
 MetricNormalizer.prototype.loadFromConfig = function loadFromConfig() {
+  const ctx = this
```

Comment:
I'd just use `this` and invoke the function with `apply` in order to save ourselves one more variable to think about, but that's just me.

Reply:
Ordinarily, I'd reach for `.bind(this)`, and `.apply` is similar, but context is still getting lost in the `forEach` with those. I'm not super enthusiastic about passing it in as an argument, but it's proving to be more reliable than `bind` or `apply` here.